File: main/java/org/goafabric/core/data/controller/PractitionerController.java
package org.goafabric.core.data.controller;

import com.oracle.svm.core.annotate.Delete;
import jakarta.validation.Valid;
import jakarta.ws.rs.*;
import jakarta.ws.rs.core.MediaType;
import org.goafabric.core.data.controller.vo.Practitioner;
import org.goafabric.core.data.logic.PractitionerLogic;

import java.util.List;

@Path("/patients")
@Produces(MediaType.APPLICATION_JSON)
public class PractitionerController {
    private final PractitionerLogic logic;

    public PractitionerController(PractitionerLogic logic) {
        this.logic = logic;
    }

    @Delete
    @Path("deleteById/{id}")
    public void deleteById(@PathParam("id") String id) {
        logic.deleteById(id);
    }

    @GET
    @Path("getById/{id}")
    public Practitioner getById(@PathParam("id") String id) {
        return logic.getById(id);
    }

    @GET
    @Path("findByGivenName")
    public List<Practitioner> findByGivenName(@PathParam("givenName") String givenName) {
        return logic.findByGivenName(givenName);
    }

    @GET
    @Path("findByFamilyName")
    public List<Practitioner> findByFamilyName(@PathParam("familyName") String familyName) {
        return logic.findByFamilyName(familyName);
    }
    
    @POST
    @Path("save")
    @Consumes(MediaType.APPLICATION_JSON)
    public Practitioner save(@Valid Practitioner patient) {
        return logic.save(patient);
    }
    
}

File: src/main/kotlin/org/goafabric/core/organization/controller/PractitionerController.kt
package org.goafabric.core.organization.controller

import jakarta.ws.rs.*
import jakarta.ws.rs.core.MediaType
import org.goafabric.core.organization.controller.dto.Practitioner
import org.goafabric.core.organization.logic.PractitionerLogic

@Path("/practitioners")
@Produces(MediaType.APPLICATION_JSON)
class PractitionerController(private val logic: PractitionerLogic) {

    @DELETE
    @Path("deleteById/{id}")
    fun deleteById(@PathParam("id") id: String) {
        logic.deleteById(id)
    }

    @GET
    @Path("getById/{id}")
    fun getById(@PathParam("id") id: String): Practitioner = logic.getById(id)

    @GET
    @Path("findByGivenName")
    fun findByGivenName(@QueryParam("givenName") givenName: String?): List<Practitioner> =
        logic.findByGivenName(givenName ?: "")

    @GET
    @Path("findByFamilyName")
    fun findByFamilyName(@QueryParam("familyName") familyName: String?): List<Practitioner> =
        logic.findByFamilyName(familyName ?: "")

    @POST
    @Path("save")
    @Consumes(MediaType.APPLICATION_JSON)
    fun save(practitioner: Practitioner): Practitioner = logic.save(practitioner)
}
